examples/energy-management-app/energy-management-common/include/DeviceEnergyManagementDelegateImpl.h

```diff
@@ -18,69 +18,276 @@
+    // interface.
+    //
+    // The rationale for this is as follows:
+    // It is envisioned there will be one master forecast object declared in DeviceEnergyManagementDelegate. When
+    // constructed, the field DataModel::List<const Structs::SlotStruct::Type> slots will be empty.
+    //
+    // The EVSEManufacturerImpl class (examples/energy-management-app/energy-management-common/include/EVSEManufacturerImpl.h) is
+    // an example implementation that a specific vendor can use as a template. It understands how the underlying energy appliance
+    // functions. EVSEManufacturerImpl inherits from DEMManufacturerDelegate
+    // (examples/energy-management-app/energy-management-common/include/DEMManufacturerDelegate.h) which is a generic interface
+    // and how the DeviceEnergyManagementDelegate class
+    // (examples/energy-management-app/energy-management-common/src/DeviceEnergyManagementDelegateImpl.cpp) communicates from the
+    // generic cluster world to the specific appliance implementation (EVSEManufacturerImpl).
+    //
+    // EVSEManufacturerImpl understands the slot structures of the appliance and configures the slot structures as follows:
+    //
+    //      Call DeviceEnergyManagementDelegate::GetForecast() to get the current forecast
+    //      Modify the slot structure - the slots memory is owned by EVSEManufacturerImpl
+    //      Call DeviceEnergyManagementDelegate::GetForecast() to set the current forecast
+    //
+    //
+    // The cluster object DeviceEnergyManagement::Instance
+    // (src/app/clusters/device-energy-management-server/device-energy-management-server.cpp) only reads the slots field of
+    // forecast when checking commands (indeed it does not modify any forecast fields itself). The DeviceEnergyManagementDelegate
+    // object does modify some of forecast's fields but does NOT modify the slots field. The only command that can modify the
+    // slots field is HandleModifyForecastRequest. Whilst DeviceEnergyManagementDelegate::ModifyForecastRequest does some state
+    // checking, the slots field is only modified by the EVSEManufacturerImpl object via the call
+    // DEMManufacturerDelegate::HandleModifyForecastRequest. DEMManufacturerDelegate::HandleModifyForecastRequest may
+    // delete/allocate the slots memory but this will be done atomically in the call to
+    // DEMManufacturerDelegate::HandleModifyForecastRequest so the underlying memory is coherent => the call to
+    // DEMManufacturerDelegate::HandleModifyForecastRequest cannot be interrupted by any other CHIP task activity.
     DataModel::Nullable<Structs::ForecastStruct::Type> mForecast;
-    // Default to NoOptOut
-    OptOutStateEnum mOptOutState = OptOutStateEnum::kNoOptOut;
+
+    // Keep track whether a PowerAdjustment is in progress
+    bool mPowerAdjustmentInProgress;
+
+    // Keep track of when that PowerAdjustment started
+    uint32_t mPowerAdjustmentStartTime;
```

Comment:
could a `bool + value` pair be tracked via a std::optional instead? Or does PowerAdjustmentStartTime make sense even if mPowerAdjustmentInProgress is false? Same question for the pause request.

Reply:
PowerAdjustmentStartTime only makes sense when mPowerAdjustmentInProgress is true. It feels clearer to me to have two separate variables tracking this rather than using a std::optional (which would work I believe). 

And it turns out I never do anything with the mPauseRequestStartTime variable so I have deleted it!